.gitignore

```diff
@@ -1,43 +1,27 @@
-# .
-/eclipse-build
-/maven-build
-/classes
-build
-/idea-build
-dist
-lib
-test-lib
-*~
-.#*
-/build.properties
-/.idea
-lucene/**/*.iml
-parent.iml
-*.ipr
-*.iws
-/*.iml
-.project
-.classpath
-.settings
-/.caches
-/prj.el
-bin
-/bin
-/bin.*
-pom.xml
-/nbproject
-/nb-build
-.pydevproject
-__pycache__
-/dev-tools/scripts/scripts.iml
-.DS_Store
-
-build/
+# Gradle
+#    Ignore the generated local settings file.
+/gradle.properties
 .gradle/
-.idea/
+build/
 
-# Ignore the generated local settings file.
-gradle.properties
+# IntelliJ IDEA
+/.idea/
+#    IntelliJ creates this folder, ignore.
```

Comment:
This is caused by the fact that it's a composite build and we don't configure it with such extra care as we do with other projects (we don't override intellij's defaults). I don't think this matters.